lib/interop.js

```diff
@@ -0,0 +1,30 @@
+class CustomElement extends HTMLElement {
+  constructor() { 
+    super(); 
+    this.dartObject = null;
+  }
+
+  connectedCallback() {
+    this.dartObject.connected();
+  }
+
+  disconnectedCallback() {
+    this.dartObject.disconnected();
+  }
+
+  attributeChangedCallback(attr, oldVal, newVal) {
+    this.dartObject.attributeChanged(attr, oldVal, newVal);
+  }
+}
+window.CustomElement = CustomElement;
+
+function defineElement(name, construct, observed) {
+  customElements.define(name, class extends CustomElement {
+    constructor() {
+      super();
+      construct(this);
```

Comment:
First problem this hits in `dart2js` is that its not a function at this point. If you change it locally to `construct.call$1` it gets further.